docs/specification.md

```diff
@@ -271,90 +270,21 @@ Parameter (in Michelson):
 - Since the contract supports only a single token type, all `token_id` values MUST be 0.
   They are passed because FA2 requires that.
 
-### **total_supply**
+### **token_metadata_registry**
 
 Types
 ```
-token_id = nat
-
-total_supply_response =
-  ( token_id :token_id
-  , nat :total_supply
-  )
-
-total_supply =
-  ( list token_id :token_ids
-  , contract (list total_supply_response) :callback
-  )
+token_metadata_registry is contract (address)
 ```
 
 Parameter (in Michelson)
 ```
-(pair %total_supply
-  (list %token_ids nat)
-  (contract %callback
-    (list
-      (pair
-        (nat %token_id)
-        (nat %total_supply)
-      )
-    )
-  )
-)
-```
-
-- This entrypoint MUST follow the FA2 requirements.
-
-- Since the contract supports only a single token type, all `token_id` values MUST be 0.
-  They are passed because FA2 requires that.
-
-### **token_metadata**
-
-Types
+(contract %token_metadata_registry address)
 ```
-token_id = nat
 
-token_metadata_response =
-  ( token_id             :token_id
-  , string               :symbol
-  , string               :name
-  , nat                  :decimals
-  , map (string, string) :extras
-  )
-
-token_metadata =
-  ( list token_id                           :token_ids
-  , contract (list token_metadata_response) :callback
-  )
-```
-
-Parameter (in Michelson)
-```
-(pair %token_metadata
-  (list %token_ids nat)
-  (contract %callback
-    (list
-      (pair
-        (nat %token_id)
-        (pair
-          (string %symbol)
-          (pair
-            (string %name)
-            (pair
-              (nat %decimals)
-              (map %extras string string)
-      ))))
-    )
-  )
-)
-```
-
-- This entrypoint MUST follow the FA2 requirements.
-
-- Since the contract supports only a single token type, all `token_id` values MUST be 0.
-  They are passed because FA2 requires that.
+- Return contract address that holds token metadata.
 
-- Token metadata for 0 `token_id` is constant and should be hardcoded in contract code or put into storage during origination.
+- Since the contract owns its token metadata the returned value of this entrypoint call will always be equal to `SELF`.
```

Comment:
I guess we should write somewhere that this contract (i.e. our stablecoin contract) must have a `big_map` with a certain annotation and type.